### PR TITLE
fix: LCM panicked due to overflow

### LINKED ISSUE #11104: `LCM` panicked due to overflow
### Describe the bug

thread 'main' panicked at /datafusion/functions/src/math/lcm.rs:85:27: 
attempt to multiply with overflow

https://github.com/apache/datafusion/blob/c7ac8b8221229b307bba434390ce1d19b7fb9358/datafusion/functions/src/math/lcm.rs#L85

### To Reproduce

```sh
DataFusion CLI v39.0.0

> select lcm(2, 9223372036854775803);
thread 'main' panicked at datafusion/functions/src/math/lcm.rs:85:27:
attempt to multiply with overflow
```

### Expected behavior

_No response_

### Additional context

Depends on #11099

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

Closes #11104.

## Rationale for this change
- Use `checked_mul` to detect multiplication overflow. 
- This PR also fixed the issue of LCM potentially returning a negative result.
```sh
DataFusion CLI v39.0.0

> select lcm(1, -9223372036854775808);
+-------------------------------------------+
| lcm(Int64(1),Int64(-9223372036854775808)) |
+-------------------------------------------+
| -9223372036854775808                      |
+-------------------------------------------+
1 row(s) fetched.
```

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/sqllogictest/test_files/scalar.slt [hunk 1]
```diff
@@ -493,41 +493,6 @@ select gcd(a, b), gcd(c, d), gcd(e, f) from signed_integers;
 2 1 1
 NULL NULL NULL
 
-## lcm
```
Comment (jonahgao) [inline datafusion/sqllogictest/test_files/scalar.slt:496]: The tests for lcm existed in two slts and have now been moved to math.slt.

File: datafusion/functions/src/math/lcm.rs [hunk 2]
```diff
@@ -75,19 +75,23 @@ impl ScalarUDFImpl for LcmFunc {
...
-            Err(_) => Err(arrow_datafusion_err!(ArrowError::ComputeError(format!(
-                "Signed integer overflow in LCM({x}, {y})"
-            )))),
-        }
+
+        // lcm(x, y) = |x| * |y| / gcd(|x|, |y|)
+        let a = x.unsigned_abs();
+        let b = y.unsigned_abs();
+        let gcd = unsigned_gcd(a, b);
+        // gcd is not zero since both a and b are not zero, so the division is safe.
```
Comment (alamb) [inline datafusion/functions/src/math/lcm.rs:86]: 👍

**Review by alamb — APPROVED** [review]
Thank you @jonahgao  -- the code looks good to me as do the tests. Nice work

**Comment by jonahgao** [comment]
Thanks @alamb for the review. Merged it since future PRs might conflict on math.slt.

**Pull request merged**